packages/core/src/textures/TextureSystem.ts

```diff
@@ -301,46 +301,273 @@ export class TextureSystem implements ISystem
     {
         glTexture.internalFormat = texture.format;
         glTexture.type = texture.type;
+
         if (this.webGLVersion !== 2)
         {
             return;
         }
 
         const gl = this.renderer.gl;
 
-        if (texture.type === gl.FLOAT)
+        if (texture.type === TYPES.UNSIGNED_BYTE)
```

Comment:
This is crazy verbose. Can we do a map/dictionary here instead?
```js
glTexture.internalFormat = glTypeFormatLookup[texture.type][texture.format];
```

Reply:
Changed it. Let me know if you like it like this.